### PR TITLE
ARROW-2353: [CI] Check correctness of built wheel on AppVeyor

### PR DESCRIPTION
**Pull request by pitrou**
Also allows building PyArrow with another generator than Visual Studio, and assorted fixes on Windows.

### CODE REVIEW

File: ci/msvc-build.bat [hunk 1]
```diff
@@ -103,12 +103,12 @@ cmake -G "%GENERATOR%" ^
 cmake --build . --target install --config %CONFIGURATION%  || exit /B
 
 @rem Needed so python-test.exe works
-set OLD_PYTHONPATH=%PYTHONPATH%
-set PYTHONPATH=%CONDA_PREFIX%\Lib;%CONDA_PREFIX%\Lib\site-packages;%CONDA_PREFIX%\python35.zip;%CONDA_PREFIX%\DLLs;%CONDA_PREFIX%;%PYTHONPATH%
+set OLD_PYTHONHOME=%PYTHONHOME%
+set PYTHONHOME=%CONDA_PREFIX%
```
Comment (pitrou) [inline ci/msvc-build.bat]: @wesm This can also be done in `python-test.exe` instead. Which one do you prefer?

**Comment by wesm** [comment]
Will review. Note this only helps nominally with the current packaging problems a little bit because the build recipe in https://github.com/apache/arrow-dist/blob/master/scripts/python-wheels-windows.bat is independent from the scripts here

**Review by xhochy — APPROVED** [review]
+1, very much in favor of this.

**Review by maxhora — APPROVED** [review]
Looks good to me! Thanks!

**Pull request closed without merging**